clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -326,7 +326,7 @@ public void poll(long timeout) throws IOException {
                     if (e instanceof EOFException || e instanceof ConnectException)
                         log.debug("Connection {} disconnected", desc);
                     else
-                        log.warn("Error in I/O with connection to {}", desc, e);
+                        log.debug("Error in I/O with connection to {}", desc, e);
```

Comment:
Does it still make sense to have the if/else here? we log debug anyway and the exception should contain enough information to figure out the type of error.